### ISSUE TITLE
Some UMD wrappers are not detected

### ISSUE THREAD

**Issue by Deraen** [google/closure-compiler#2586]
I've hit problems with a few libraries which use UMD wrappers which Closure doesn't detect:

Leaflet https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.1.0/leaflet-src.js:
```js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
	typeof define === 'function' && define.amd ? define(['exports'], factory) :
	(factory((global.L = global.L || {})));
}(this, (function (exports) {
   exports.foo = "bar";
})));
```

Bowser https://github.com/lancedikson/bowser/blob/master/src/bowser.js:
```js
!function (root, name, definition) {
  if (typeof module != 'undefined' && module.exports) module.exports = definition()
  else if (typeof define == 'function' && define.amd) define(name, definition)
  else root[name] = definition()
}(this, 'bowser', function () {
    return {foo: 'bar'};
});
```

**Comment by ChadKillingsworth** [comment]
There are so many of these, it's very hard to properly detect. It'll probably be quite some time before I can look into this.

**Comment by ChadKillingsworth** [comment]
Most of these variations are now properly rewritten.
